### PR TITLE
Fast GC compacted dropped segment

### PR DESCRIPTION
**Pull request by czs007**
#26205

### CODE REVIEW

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: internal/datacoord/garbage_collector.go [hunk 1]
```diff
@@ -202,6 +202,51 @@ func (gc *garbageCollector) scan() {
...
+	cpTimestamp Timestamp) bool {
+
+	hasChild := childSegment != nil
+
+	log := log.With(zap.Int64("segmentID", segment.ID))
+	isCompacted := hasChild || segment.GetCompacted()
+
+	if isCompacted {
+		if !hasChild {
+			log.Info("trigger fast GC segment")
```
Comment (xiaofan-luan) [inline internal/datacoord/garbage_collector.go:217]: should be logged with more information

File: internal/datacoord/garbage_collector_test.go [hunk 2]
```diff
@@ -1003,6 +1052,25 @@ func TestGarbageCollector_clearETCD(t *testing.T) {
...
 
+	/*
+		A    B
+		 \   /
+		   C	D
+		    \  /
+		      E
+
+		E: flushed, not indexed, should not be GCed
+		D: dropped, not indexed, should not be GCed, since E is not GCed
```
Comment (czs007) [inline internal/datacoord/garbage_collector_test.go:1063]: since E is not indexed

**Pull request merged**